pkg/chains/formats/slsa/v2alpha2/internal/build_definitions/taskrun.go

```diff
@@ -0,0 +1,65 @@
+package builddefinitions
+
+import (
+	"fmt"
+
+	"github.com/tektoncd/chains/pkg/chains/objects"
+)
+
+type TaskBuildType struct {
+	BuildType          string
+	Tro                *objects.TaskRunObject
+	InternalParameters func(tro *objects.TaskRunObject) map[string]any
+}
+
+// internalParameters adds the tekton feature flags that were enabled
+// for the taskrun.
+func SLSATaskInternalParameters(tro *objects.TaskRunObject) map[string]any {
+	internalParams := make(map[string]any)
+	if tro.Status.Provenance != nil && tro.Status.Provenance.FeatureFlags != nil {
+		internalParams["tekton-pipelines-feature-flags"] = *tro.Status.Provenance.FeatureFlags
+	}
+	return internalParams
+}
+
+func TektonTaskInternalParameters(tro *objects.TaskRunObject) map[string]any {
+	internalParams := make(map[string]any)
+	if tro.Status.Provenance != nil && tro.Status.Provenance.FeatureFlags != nil {
+		internalParams["tekton-pipelines-feature-flags"] = *tro.Status.Provenance.FeatureFlags
+	}
+	internalParams["labels"] = tro.ObjectMeta.Labels
+	internalParams["annotations"] = tro.ObjectMeta.Annotations
+	return internalParams
+}
+
+func (t TaskBuildType) GetInternalParameters() map[string]any {
+	return t.InternalParameters(t.Tro)
+}
+
+// externalParameters adds the task run spec
+func (t TaskBuildType) GetExternalParameters() map[string]any {
+	externalParams := make(map[string]any)
+	// add origin of the top level task config
+	// isRemoteTask checks if the task was fetched using a remote resolver
+	isRemoteTask := false
+	if t.Tro.Spec.TaskRef != nil {
+		if t.Tro.Spec.TaskRef.Resolver != "" && t.Tro.Spec.TaskRef.Resolver != "Cluster" {
+			isRemoteTask = true
+		}
+	}
+	if t := t.Tro.Status.Provenance; t != nil && t.RefSource != nil && isRemoteTask {
```

Comment:
We could do something with the TektonObject interface here as well to avoid duplication.

Reply:
done!